File: lab_solutions/sol_lab2task3/src/publisher_node.cpp
#include "ros/ros.h"
//TODO: include the library for a msg typed VectorStamped
#include "geometry_msgs/Vector3Stamped.h"

int main(int argc, char **argv)
{

    ros::init(argc, argv, "talker");

    //TODO: Define a nodehandle and a publisher
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<geometry_msgs::Vector3Stamped>("/publish_point", 1000);

    ros::Rate loop_rate(10);

    double radius = 1.0;
    double period = 5.0;

    //TODO: Define your message
    geometry_msgs::Vector3Stamped msg;

    double starting_time =ros::Time::now().toSec();

    while (ros::ok())
    {

        //TODO: Input the data into your message. Your message should be a position of a unit circle on the XZ-plane.
        msg.header.stamp = ros::Time::now();

        msg.vector.x = radius * cos(2 * M_PI * (msg.header.stamp.toSec() - starting_time)/period);
        msg.vector.y = 0;
        msg.vector.z = radius * sin(2 * M_PI * (msg.header.stamp.toSec() - starting_time)/period);

        pub.publish(msg);

        ros::spinOnce();

        loop_rate.sleep();
    }


    return 0;
}

File: lab02/lab02task03/scripts/publisher_node.py
#!/usr/bin/env python

import rospy
import math
##TODO: include the library for a msg typed VectorStamped

def talker():

    ##TODO: Define a publisher
    pub = rospy.Publisher('chatter', PointStamped, queue_size=10)
    rospy.init_node('talker', anonymous=True)
    rate = rospy.Rate(10) # 10hz

    radius = 1.0
    period = 5.0

    ##TODO: Define your message

    starting_time = rospy.Time.now().to_sec()

    while not rospy.is_shutdown():

        ##TODO: Input the data into your message. Your message should be a position of a unit circle on the XZ-plane.
        msg.header.stamp = rospy.Time.now()

        msg.point.x = radius * math.cos(2 * math.pi * (msg.header.stamp.to_sec() - starting_time)/period)
        msg.point.y = radius * math.sin(2 * math.pi * (msg.header.stamp.to_sec() - starting_time)/period)
        msg.point.z = 0.0

        pub.publish(msg)

        rate.sleep()

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass
